### PR TITLE
Refactor - property.py refactor

### PR DESCRIPTION
**Pull request by emmanesbit**
Currently just contains break-out of classes into separate files. 

Further refactoring, changes and unit tests will follow in another PR.

[x] PropertyKind
[ ] GridPropertyCollection
[x] Property
[ ] property_common
[x] StringLookup
[ ] WellIntervalProperty
[ ] WellIntervalPropertyCollection
[x] WellLog
[x] WellLogCollection

### CODE REVIEW

File: resqpy/property/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    'PropertyCollection', 'Property', 'WellLog', 'WellIntervalProperty', 'WellIntervalPropertyCollection',
+    'WellLogCollection', 'StringLookup', 'PropertyKind', 'GridPropertyCollection',
+    'property_over_time_series_from_collection', 'property_collection_for_keyword', 'infer_property_kind',
+    'write_hdf5_and_create_xml_for_active_property', 'reformat_column_edges_to_resqml_format',
+    'reformat_column_edges_from_resqml_format', 'same_property_kind', 'selective_version_of_collection',
+    'supported_local_property_kind_list', 'supported_property_kind_list', 'supported_facet_type_list',
+    'expected_facet_type_dict', 'create_transmisibility_multiplier_property_kind'
+]
+
+from .property_common import property_collection_for_keyword, property_over_time_series_from_collection, write_hdf5_and_create_xml_for_active_property, infer_property_kind, reformat_column_edges_to_resqml_format, reformat_column_edges_from_resqml_format, selective_version_of_collection, same_property_kind, supported_property_kind_list, supported_local_property_kind_list, supported_facet_type_list, expected_facet_type_dict
```
Comment (andy-beer) [inline resqpy/property/__init__.py:11]: please split this very long line, either using a \ continuation character or simply repeating the from..import

File: resqpy/property/gridpropertycollection.py [hunk 2]
```diff
@@ -0,0 +1,1044 @@
+"""property.py: module handling collections of RESQML properties for grids, wellbore frames, grid connection sets etc."""
```
Comment (andy-beer) [inline resqpy/property/gridpropertycollection.py:1]: Module level docstrings will need to be made specific to each new module

File: resqpy/property/wellintervalpropertycollection.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+"""property.py: module handling collections of RESQML properties for grids, wellbore frames, grid connection sets etc."""
```
Comment (andy-beer) [inline resqpy/property/wellintervalpropertycollection.py:1]: These 4 well*.py modules could be kept as one, maybe?

**Review by andy-beer — COMMENTED** [review]
Personally, I have a very strong preference for using _ in file names, rather than running words together, so property_collection.py rather than propertycollection.py

**Review by andy-beer — APPROVED** [review]

**Comment by andy-beer** [comment]
@emmanesbit Please check that all public methods and functions in the property package have a docstring.

**Comment by emmanesbit** [comment]
@cflynn3, @andy-beer - To make it easier to review the changes later on (as these files can get quite large), @connortann has suggested merging this PR now with just the 'splitting', then making another PR for refactoring to make the changes easier to understand.

This PR does contain some minor changes to property_collection, but not too many. Broadly this PR just contains splitting of the files.

Ready to review when you are all happy.

**Review by andy-beer — APPROVED** [review]
I'm happy for the changes so far to be merged.

**Review by andy-beer — APPROVED** [review]

**Review by cflynn3 — APPROVED** [review]

**Pull request merged**